lb_content_resolver/subsonic.py

```diff
@@ -135,16 +135,45 @@ def update_recordings(self, recordings):
             Updates recording_id, subsonic_id, last_update
         """
 
-        recordings = [(r[0], r[1], datetime.datetime.now()) for r in recordings]
+        recording_index = { r[0]:r[1] for r in recordings }
 
         cursor = db.connection().cursor()
         with db.atomic() as transaction:
-            cursor.executemany(
-                """INSERT INTO recording_subsonic (recording_id, subsonic_id, last_updated)
-                                        VALUES (?, ?, ?)
-                     ON CONFLICT DO UPDATE SET recording_id = excluded.recording_id
-                                             , subsonic_id = excluded.subsonic_id
-                                             , last_updated = excluded.last_updated""", recordings)
+
+            placeholders = ",".join(("?", ) * len(recording_index.keys()))
+            cursor.execute("""SELECT recording_id
+                                FROM recording_subsonic
+                               WHERE recording_id in (%s)""" % placeholders, tuple(recording_index.keys()))
+            existing_ids = { row[0]:None for row in cursor.fetchall() }
```

Comment:
Is there a specific reason to use full sql here instead of letting peewee generate it?

I mean something like (totally untested):

```
RecordingSubsonic.select(RecordingSubsonic.recording_id).where(Recording.file_recording_id.in_(tuple(recording_index))).tuples()
```

Also you don't need to build placeholders with such syntax.
Same goes for upsert below, see https://docs.peewee-orm.com/en/latest/peewee/querying.html#bulk-inserts

Not sure how it works on older setups, but perhaps peewee takes care of that.

Doesn't the code at https://github.com/metabrainz/listenbrainz-content-resolver/blob/0eecf32e0c1c2c1166a3cb733391ea85a1612053/lb_content_resolver/database.py#L282 work correctly in this rpi env?

Reply:
For multi-row and join queries, I prefer to use SQL directly. ORMs hide some crucial functionality in its whiz-bang introspective approach to things, but then never give examples of complex queries. Mainly, I think, because no one uses ORMs for complex queries. Many times I've tried to start with the ORM UI, only to waste a lot of time on it and then go with plain SQL. Cutting out the wasted hours and just going SQL here, which is my preference really.